src/rules/selector-max-depth/README.md

```diff
@@ -0,0 +1,54 @@
+# selector-max-depth
+
+Limit the depth of selectors.
+
+```css
+   .foo .bar[data-val] > .baz + .boom > .lorem {
+/* ↑    ↑                 \_________/   ↑  
+   |    |                  ↑            | 
+  Lv1  Lv2                Lv3          Lv4  -- these are depth levels */
+```
+
+To put it simply, a selector's depth in terms of this rule is how many different levels of HTML structure (not necessarily direct descendants) the selector is reflecting. You can read more about selectors depth is [SMACSS book](http://smacss.com/book/applicability). Although note, that this rule is not about the actual number of HTML levels, since it is usually hard to say how many elements wrap the `a` in `body a` without looking in the markup.
+
+Only child (`h1 > a`) and descendant (`h1 a`) combinators are considered to create a new depth level; adjacent combinators (`p + p`, `.foo ~ bar`) don't.
```

Comment:
> I thought you changed this.
Not yet, was waiting for the decision on selector-parser vs performance before applying all the changes.